File: src/domain/HomeDomain.ts
import { DesignSystemMetadataHome } from "./DesignSystemDomain";
import { PaletteBuilderMetadata } from "./PaletteBuilderDomain";

export type InsertFilePayload = {
  filePath: string;
};

export type RecentFilesMetadata =
  | { DesignSystem: DesignSystemMetadataHome }
  | { PaletteBuilder: PaletteBuilderMetadata }
  | { Unknown: string };

export type RemoveRecentFilesPayload = {
  filePath: string;
  isDeleteFromComputer: boolean;
};

export interface RecentFile {
  filePath: string;
  editMode: string;
  category: RecentFileCategory;
}

export type RecentFileCategory =
  | "PaletteBuilderCategory"
  | "DesignSystemCategory";

export interface PresetDressing {
  banners: string[];
  logos: string[];
}

export interface UserSettings {
  pluginDisplayMode: PluginDisplayMode;
}

export type PluginDisplayMode = "fullscreen" | "modal";

File: src-tauri/src/domain/home_domain.rs
use std::path::PathBuf;

use super::{
    design_system_domain::DesignSystemMetadataHome, palette_builder_domain::PaletteBuilderMetadata,
};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub enum RecentFilesMetadata {
    DesignSystem(DesignSystemMetadataHome),
    PaletteBuilder(PaletteBuilderMetadata),
    Unknown(PathBuf),
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum RecentFileCategory {
    DesignSystemCategory,
    PaletteBuilderCategory,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]

pub struct RemoveRecentFilesPayload {
    pub file_path: PathBuf,
    pub is_delete_from_computer: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct RecentFile {
    pub file_path: PathBuf,
    pub edit_mode: Option<bool>,
    pub category: RecentFileCategory,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PresetDressing {
    pub banners: Vec<String>,
    pub logos: Vec<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct UserSettings {
    pub plugin_display_mode: PluginDisplayMode,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub enum PluginDisplayMode {
    #[serde(rename = "fullscreen")]
    FullScreen,
    #[serde(rename = "modal")]
    Modal,
}

impl Default for UserSettings {
    fn default() -> Self {
        UserSettings {
            plugin_display_mode: PluginDisplayMode::FullScreen,
        }
    }
}
